src/tfr_aruco/src/tfr_aruco_server.cpp

```diff
@@ -0,0 +1,144 @@
+        
+        // add rest of marker from the top left points (all the same size)
+        for(auto &marker : boardCorners) 
+        {
+            marker.push_back(marker[0] + cv::Point3f(3.2, 0,   0));
+            marker.push_back(marker[0] + cv::Point3f(3.2, 3.2, 0));
+            marker.push_back(marker[0] + cv::Point3f(0,   3.2, 0));
+            
+            // from cm to meters
+            for(auto &p : marker) {
+                p *= .01;
+            }
+        }
+        std::vector<int> boardIds = {1, 2, 3, 4, 5, 6, 7, 8};
+        
+        board = cv::aruco::Board::create(std::move(boardCorners), dictionary, std::move(boardIds));
+
+        // set up params
+        params = cv::Ptr<cv::aruco::DetectorParameters>(new cv::aruco::DetectorParameters);
+        params->cornerRefinementMethod = cv::aruco::CORNER_REFINE_SUBPIX;
+        //params->cornerRefinementMethod = cv::aruco::CORNER_REFINE_CONTOUR;
+        params->cornerRefinementWinSize = 10;
+    }
+    // This is the method that will be called when a client makes use
+    // of this server. The provided goal is the "input".
+    void execute(const tfr_msgs::ArucoGoalConstPtr& goal, Server* server)
+    {
+        if (server->isPreemptRequested() || !ros::ok())
+        {
+            server->setPreempted();
+            return;
+        }
+        cameraModel.fromCameraInfo(goal->camera_info);
+    
+    
+        // convert ROS message to opencv image
+        // the image is stored at imageHolder->image
+        cv_bridge::CvImagePtr imageHolder;
+        try 
+        {
+            imageHolder = cv_bridge::toCvCopy(goal->image, sensor_msgs::image_encodings::BGR8);
+        }
+        catch (cv_bridge::Exception& e)
+        {
+            ROS_ERROR("cv_bridge exception: %s", e.what());
+            return;
+        }
+    
+    
+        // detect fiducial markers
+        std::vector<int> markerIds;
+        std::vector<std::vector<cv::Point2f> > markerCorners;
+        
+        cv::aruco::detectMarkers(imageHolder->image, dictionary, markerCorners, markerIds, params);
+    
+    
+        // get individual marker poses
+        cv::Mat cameraMatrix = cv::Mat(cameraModel.fullIntrinsicMatrix()).clone();
+        cv::Mat distCoeffs = cameraModel.distortionCoeffs().clone();
+        
+        std::vector< cv::Vec3d > rvecs, tvecs;
+        cv::aruco::estimatePoseSingleMarkers(markerCorners, 0.05, cameraMatrix, distCoeffs, rvecs, tvecs);
+    
+    
+    
+        
+        cv::Vec3d boardRotVec, boardTransVec;
+        int markersDetected = cv::aruco::estimatePoseBoard(markerCorners, markerIds, board, cameraMatrix, distCoeffs, boardRotVec, boardTransVec);
+        
+        tfr_msgs::ArucoResult result;
+        result.number_found = markersDetected;
+        result.position = std::vector<float>(3, 0); // 3 members initialized to 0
+        result.rotation = std::vector<float>(3, 0); // 3 members initialized to 0
```

Comment:
You can eliminate the zero inititialization step by just doing this, not required, but cleaner.
 `tfr_msgs::ArucoResult result{};`
`//... fill result with stuff`